node_modules/oae-core/signin/js/signin.js

```diff
@@ -0,0 +1,196 @@
+/*!
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['jquery','oae.core'], function($, oae) {
+
+    return function(uid, showSettings, widgetData) {
+
+        // The widget container
+        var $rootel = $('#' + uid);
+
+        var authStrategyInfo = null;
+        var signInRedirectUrl = null;
+        var invitationInfo = null;
+        var showSignUp = true;
+
+        /**
+         * Finish the login process by showing the correct validation message in case of a failed
+         * login attempt, or by redirecting the user in case of a successful login attempt
+         *
+         * @param  {Error}      err        Error object containing error code and error message
+         */
+        var finishLogin = function(err) {
+            if (err) {
+                var signInMessage = oae.api.i18n.translate('__MSG__SIGN_IN__');
+                $('#signin-local-form button', $rootel).html(signInMessage);
+                $('#signin-local-form *', $rootel).prop('disable', false);
+
+                // Set a `signin-local-failed-attempt` on the fields to tell jquery.validate
+                // that the field is invalid
```

Comment:
I'm not getting flickering, but I do see it complaining about a message for "username":

![image](https://cloud.githubusercontent.com/assets/102265/9865612/cb908dbc-5b26-11e5-9461-091e15045d93.png)

Which I've fixed. FYI that screencast just showed a black screen with an arrow flicking around. It was really weird. I think I'll need a verbal description instead